projects/packages/publicize/src/class-publicize-setup.php

```diff
@@ -22,18 +22,52 @@ public static function configure() {
 	 * To configure the publicize package, when called via the Config package.
 	 */
 	public static function on_jetpack_feature_publicize_enabled() {
-
 		global $publicize_ui;
+
 		if ( ! isset( $publicize_ui ) ) {
 			$publicize_ui = new Publicize_UI();
 
 		}
+
 		// Adding on a higher priority to make sure we're the first field registered.
 		// The priority parameter can be removed once we deprecate WPCOM_REST_API_V2_Post_Publicize_Connections_Field
 		add_action( 'rest_api_init', array( new Connections_Post_Field(), 'register_fields' ), 5 );
 
 		add_action( 'rest_api_init', array( new REST_Controller(), 'register_rest_routes' ) );
+
+		add_action( 'current_screen', array( static::class, 'init_sharing_limits' ) );
 	}
-}
 
+	/**
+	 * Initialise share limits if they should be enabled.
+	 */
+	public static function init_sharing_limits() {
+		$current_screen = get_current_screen();
+
+		if ( empty( $current_screen ) || $current_screen->base !== 'post' ) {
+			return;
+		}
+
+		global $publicize;
 
+		if ( $publicize->has_paid_plan() ) {
+			return;
+		}
+
+		$info = $publicize->get_publicize_shares_info( \Jetpack_Options::get_option( 'id' ) );
```

Comment:
Two more ponderings here.

- It would be good to re-use this init function on WPCOM, and so we'd need to use `get_current_blog_id()` here. As we probably wouldn't be calling it on the `current_screen` hook, we could maybe pass the blog ID in, and if it's null/0 default to the Jetpack_Options?
- `get_publicize_shares_info` will make an API call everytime. As we roll this out, is there a way to cache the feature flag value locally (when it's false and when we don't receive the shares info, as that shows we're not interested in it)? If we store it in a transient for a long time, we could invalidate it on [the `jetpack_get_site_data_success` hook](
https://github.com/Automattic/jetpack/tree/d5dcf6fab60220ed844d52f162c6bd9c7bac5bca/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#L1935), so it gets refreshed each time the next time after the site data has been refreshed.